### PR TITLE
Add automation properties to expander-style settings

### LINKED ISSUE #13019: [Settings - Startup]: Name is not defined for the button present under 'Startup' tab.
### Windows Terminal version

1.13.10984.0

### Windows build number

22598.200

### Other Software

**Test Environment:**
**OS:** Windows 11 Version 21H2 (OS Build 22598.200)
**Screen Reader:** Narrator

### Steps to reproduce

**Repro Steps:**
Open Windows Terminal.
Open Settings page using "Ctrl + ',' ".
Verify that Name is not defined for the button present under 'Startup' tab.

**User Experience: -> Priority 1**
User who relies on screen reader will be affected and not be able to know what the control is and what is its functionality if name is not defined for the control.

**Guideline Reference:**
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

**Attachments:**
[Name is not defined for the button present under 'Startup' tab..zip](https://github.com/microsoft/terminal/files/8602787/Name.is.not.defined.for.the.button.present.under.Startup.tab.zip)

### Expected Behavior

Name should be defined for the button present under 'Startup' tab.
Screen reader should announce "When terminal starts button collapse/expand button".

### Actual Behavior

Name is not defined for the button present under 'Startup' tab.
Screen reader announces, "group collapse".

### PR DESCRIPTION
**Pull request by PankajBhojwani**
## Summary of the Pull Request
Make sure we set `Name` and `FullDescription` on expander-style settings in the SUI

## PR Checklist
* [x] Closes #13019 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [x] I work here

## Validation Steps Performed
Accessibility insights now shows the name/full description for the expander-style settings

### CODE REVIEW

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp [hunk 1]
```diff
@@ -146,6 +146,30 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
 
+        if (const auto& child{ GetTemplateChild(L"Expander") })
+        {
+            if (const auto& expander{ child.try_as<Microsoft::UI::Xaml::Controls::Expander>() })
+            {
+                // apply header text as name (automation property)
+                if (const auto& header{ Header() })
+                {
+                    const auto headerText{ header.try_as<hstring>() };
+                    if (headerText && !headerText->empty())
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:157]: I believe an empty hstring is the same as a null hstring, so you can just put this all in one condition?

```
if (const auto headerText{ header.try_as<hstring>() })
```

(same applies below!)

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp [hunk 2]
```diff
@@ -146,6 +146,30 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             }
         }
 
+        if (const auto& child{ GetTemplateChild(L"Expander") })
+        {
+            if (const auto& expander{ child.try_as<Microsoft::UI::Xaml::Controls::Expander>() })
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:151]: This is just to ensure we have the correct control, right? If there was a template child named Expander we wouldn't want to give it these properties _unless it actually was an expander_?

Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:151]: This is just to ensure we have the correct control, right? If there was a template child named Expander we wouldn't want to give it these properties _unless it actually was an expander_?

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp [hunk 3]
```diff
@@ -146,6 +146,30 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             }
         }
 
+        if (const auto& child{ GetTemplateChild(L"Expander") })
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:149]: We should probably reorder these to be...
```c++
if (/*we have an Expander*/) {/*apply Auto Props to expander control*/}
else if (/*we have Content*/) { /*apply Auto Props to content*/ }
```
That way we aren't redundantly applying the Auto Props.

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp (hunk 1 above)
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:157]: (if you make this change, please apply it in the other places we've done this in this file [or at least this function])

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp (hunk 2 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:151]: Yeah I like to think of this as just a sanity check! Would rather have it than not though

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:151]: (idk why this comment happened twice, resolving this one)

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp [hunk 4]
```diff
@@ -122,23 +122,42 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+                if (const auto& header{ Header() })
+                {
+                    if (const auto headerText{ header.try_as<hstring>() })
+                    {
+                        Automation::AutomationProperties::SetName(expander, *headerText);
+                    }
+                }
+
+                // apply help text as tooltip and full description (automation property)
+                if (const auto& helpText{ HelpText() }; !helpText.empty())
```
Comment (PankajBhojwani) [inline src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:139]: `if (const auto& helpText{ HelpText() })` does not work for some reason

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:139]: this is great then

File: src/cascadia/TerminalSettingsEditor/SettingContainer.cpp [hunk 5]
```diff
@@ -122,23 +122,42 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+                {
+                    Controls::ToolTipService::SetToolTip(expander, box_value(helpText));
+                    Automation::AutomationProperties::SetFullDescription(expander, helpText);
+                }
+            }
+        }
+        else if (const auto& content{ Content() })
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:146]: i actually think, by Carlos' suggestion, you can simplify even more. After all, the only difference between lines 134,141 and 155,162 is nothing. There's no difference...! so you can simplify the code significantly

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/SettingContainer.cpp:133]: Sounds good, added a helper

**Review by DHowett — APPROVED** [review]

**Pull request merged**